### PR TITLE
Add OAuth to application so other teams can add it

### PR DESCRIPTION
**Pull request by thelowlypeon**
This adds a new Grape API that solely handles auth stuff. It stores resulting tokens in a `Team` model, which is now linked to a user

### CODE REVIEW

**Comment by thelowlypeon** [comment]
sorry for the many updates post PR creation. required some in-the-wild testing because slack won't hit my localhost.

this is ready to go 👍

File: db/connect.rb [hunk 1]
```diff
@@ -3,6 +3,9 @@
   when :test
     require 'sqlite3'
     DB = Sequel.connect('sqlite://test/stations_bot-test.db')
+  when :development
+    require 'mysql2'
+    DB = Sequel.connect(ENV['DATABASE_URL'])
```
Comment (andygiannini) [inline db/connect.rb]: We should probably clean up these database connections, too. I won't block this PR on it, though.

File: oauth.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          team = Team.find_or_create(team_id: json['team_id'])
+          data = {access_token: json['access_token']}
+          data['team_name'] = json['team_name'] if json.key?('team_name')
+          team.update(data)
+          redirect "https://stationtostationapp.com/slack?success=true"
+        else
+          redirect "https://stationtostationapp.com/slack?success=false"
+        end
+      else
+        redirect "https://stationtostationapp.com/slack?success=false"
```
Comment (andygiannini) [inline oauth.rb]: Could we simplify this a bit to something like:

```
      success = false
      if response = RestClient.get('https://slack.com/api/oauth.access', {...})
        json = JSON.parse(response.body)
        if json.key?('access_token') && json.key?('team_id')
          ...
          success = true
        end
      end
      redirect "https://stationtostationapp.com/slack?success=#{success}"
```

**Comment by thelowlypeon** [comment]
@andygiannini yikers, this required some more work than i'd expected.

why? because i was an idiot and removed the token from `.env` so as to keep it out of the repo, but i didn't have it on my local machine. i kept getting 401 unauth, but that's because the station to station api was returning that status :(

anyway, addressed your suggestions

**Comment by thelowlypeon** [comment]
one small comment -- i keep getting this failure, but it only happens with sqlite. can you give this a pull and try testing?

```
  1) Failure:
SavePlaceCommandTest#test_saves_new_place [/Users/peter/Sites/stations-bot/test/save_place_command_test.rb:13]:
Failed to save place: {"text":"Unknown error SQLite3::SQLException: no such table: main.users_backup0"}.
Expected: 1
  Actual: 0
```

**Comment by thelowlypeon** [comment]
@andygiannini fixed that sqlite error i was getting. i forgot that sqlite doesn't support foreign key constraints, and apparently `sequel` doesn't handle it well

**Comment by andygiannini** [comment]
Nice fixes for the DB and tests! 👍

**Pull request merged**